packages/react-components/react-icons-compat/src/icon.ts

```diff
@@ -0,0 +1,222 @@
+import * as React from 'react';
+import { GlobalSettings } from './GlobalSettings';
+import { warn } from './warn';
+
+export type T = React.ReactNode;
```

Comment:
If we remove this dummy type right now, `api-extractor` complains about not being able to find the definition for the `JSX` symbol.

Reply:
This issue is tracked in #30236, will follow up in a separate PR